src/basic/strv.c

```diff
@@ -724,6 +724,11 @@ int strv_make_nulstr(char **l, char **p, size_t *q) {
                 if (!m)
                         return -ENOMEM;
                 n = 0;
+        } else {
+                /* make sure there is a second extra NUL at the end of resulting nulstr */
+                if (!GREEDY_REALLOC(m, n_allocated, n + 1))
+                        return -ENOMEM;
+                m[n] = '\0';
```

Comment:
I wondered if I should add 1 to n too ; my final opinion is that it should not be the "count of allocated memory" with the final NUL, but the "string length" and, since it is a nulstr, with the example of :

> a\0b\0

n should be 4 here, although we finally allocated 5 bytes to put the extra NUL at the end.